src/modules/catalog/catalog-directive.js

```diff
@@ -99,4 +99,15 @@ angular.module('anol.catalog')
                     });
                 } 
             };
-        });
+        })
+        // filter directive needed when we want to be able to sort the objects on the catalog.
+        // It's not possible to siort on an object
+        .filter("catalogToArray", function(){
+          return function(obj) {
+              return Object
+                .entries(obj)
+                .map(function(entry){
+                  return entry[1]
```

Comment:
Is it ok to discard `entry[0]`?

Reply:
yes, the first position in the array is just a letter. the array itself looks something like this: `["A", {title: "A", layers: [...], localeTitle: "A"}]`, so we don't care about the first position.
Anyaway, this is just a filter and it is actually not replacing the object that will be looped through.